### ISSUE TITLE
Error when session stale, re-logging in

### ISSUE THREAD

**Issue by zx900930** [PokemonGoF/PokemonGo-Bot#812]
Bot stopped when trying to re-logging in
```
[14:53:15] Session stale, re-logging in
[14:53:15] [#] Attempting login to Pokemon Go.
Traceback (most recent call last):
  File "pokecli.py", line 228, in <module>
    main()
  File "pokecli.py", line 219, in main
    bot.take_step()
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 42, in take_step
    self.work_on_cell(cell, location)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 73, in work_on_cell
    self.check_session(position)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 192, in check_session
    self.login()
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 200, in login
    self.api.set_position(*self.position)
TypeError: set_position() takes exactly 4 arguments (3 given)
```

### Other Information
OS: Win10
Git Commit: 2a15c03 ("Oopses - small mistake!", 2016-07-25)

**Comment by JonathanRosado** [comment]
+1 This happened to me after 30mins of botting.

**Comment by mehdi-S** [comment]
same here

**Comment by boriss579** [comment]
I usually do pip install command again and this fix the login problem, maybe try and it will help you also.

**Comment by mehdi-S** [comment]
@boriss579 it solve starting problem usually

**Comment by JonathanRosado** [comment]
@boriss579 Thanks. I'll try it. Although, TBH, i don't see how reinstalling the modules might fix this.

**Comment by zx900930** [comment]
@boriss579  I use a batch file to update this, it will redo the pip install OFC, but I still got the relogin error.

my `update.bat`:

```
@echo off
git fetch --all
git reset --hard origin/dev
git pull
pip install -r requirements.txt
IF EXIST ".\PoGoBot" (
cd PoGoBot
git fetch --all
git reset --hard
git pull
) ELSE (
mkdir PoGoBot
cd PoGoBot
rmdir /s /q ".\OpenPoGoWeb"
git clone https://github.com/OpenPoGo/OpenPoGoWeb.git
)
ROBOCOPY ".\OpenPoGoWeb" "..\web" /e
```

**Comment by mehdi-S** [comment]
try to git pull, it is now magically fixed for me 

`[09:51:44] Session stale, re-logging in
[09:51:44] [#] Attempting login to Pokemon Go.
[09:51:45] [+] Login to Pokemon Go successful.`

**Comment by leadboots5** [comment]
the stale session stuff was put in, but with a typo. it looks like the typo was fixed, but I can't find a record of when actually... I haven't tested it myself, but this can most likely be closed with the current dev.

**Comment by zx900930** [comment]
new error
```
`[16:56:25] Session stale, re-logging in
[16:56:25] Attempting login to Pokemon Go.
[16:56:26] Login to Pokemon Go successful.
[16:56:26] Scanning area for objects....
[16:56:26] Walking from (41.381490199999995, 2.1796107) to (41.381490199999995, 2.1796107) 0.00km
Traceback (most recent call last):
  File "pokecli.py", line 228, in <module>
    main()
  File "pokecli.py", line 219, in main
    bot.take_step()
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 39, in take_step
    cells = self.find_close_cells(*location)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 108, in find_close_cells
    self.update_web_location(map_cells,lat,lng)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 51, in update_web_location
    alt = self.position[2]
IndexError: tuple index out of range`
```
commit： fa9df3b ("Fixing the Desiigner - Panda reference during Debug.....", 2016-07-25)

**Comment by surfaace** [comment]
issue 'TypeError: set_position() takes exactly 4 arguments (3 given)' was fixed in fb2cca3b1f701653d58424f49a2ba7fae5cd97e9

tuple error is tracked in #864
